src/node_buffer.cc

```diff
@@ -1223,6 +1223,14 @@ static void EncodeInto(const FunctionCallbackInfo<Value>& args) {
   results[1] = written;
 }
 
+static void IsUtf8(const FunctionCallbackInfo<Value>& args) {
+  CHECK_GE(args.Length(), 1);
+  CHECK(args[0]->IsArrayBuffer());
+  Local<ArrayBuffer> input = args[0].As<ArrayBuffer>();
+  auto external = static_cast<const char*>(input->Data());
+  args.GetReturnValue().Set(
+      simdutf::validate_utf8(external, input->ByteLength()));
```

Comment:
It's also been pointed out above but this checks the whole arraybuffer, which isn't what you want for things that are arraybufferviews (i.e., arraybuffer slices.)